### PR TITLE
fix: Refactor isbservice_controller to eliminate use of GenericObject

### PR DESCRIPTION
**Pull request by chandankumar4**
Fixes
ISBService Controller:  https://github.com/numaproj/numaplane/issues/143

### Modifications

Removed the usages of Generic Object struct and use Unstructured.

### CODE REVIEW

File: internal/util/kubernetes/unstructured_util.go [hunk 1]
```diff
@@ -261,6 +281,12 @@ func CreateResource(ctx context.Context, c client.Client, obj *GenericObject) er
 	return c.Create(ctx, unstructuredObj)
 }
 
+// CreateResourceUnstructured creates the resource in the kubernetes cluster
+// TODO: This is a temporary function which will be removed once all the controller are migrated to use Unstructured Object
```
Comment (juliev0) [inline internal/util/kubernetes/unstructured_util.go:285]: when you say these are "temporary functions", do you mean that the logic for them will still exist, but basically they will be renamed as the original function names? (e.g. `CreateResource()`)

File: internal/util/kubernetes/unstructured_util.go [hunk 2]
```diff
@@ -51,6 +51,26 @@ func ParseStatus(obj *GenericObject) (GenericStatus, error) {
...
+
+	var status GenericStatus
+	objByte, err := json.Marshal(statusRaw)
+	if err != nil {
+		return GenericStatus{}, err
+	}
+	err = json.Unmarshal(objByte, &status)
+	if err != nil {
+		return GenericStatus{}, err
+	}
```
Comment (juliev0) [inline internal/util/kubernetes/unstructured_util.go:69]: nit: can you use the [util.StructToStruct() ](https://github.com/numaproj/numaplane/blob/main/internal/util/util.go#L11)method here?

**Review by juliev0 — APPROVED** [review]

File: internal/util/kubernetes/unstructured_util.go (hunk 2 above)
Comment (juliev0) [reply internal/util/kubernetes/unstructured_util.go:69]: I'm okay if you want to merge this PR and do this as a follow up

Comment (juliev0) [reply internal/util/kubernetes/unstructured_util.go:69]: Actually, in the interest of moving forward, I think I'll just merge this so I can merge main into my PR and resolve the conflicts.

File: internal/util/kubernetes/unstructured_util.go (hunk 1 above)
Comment (chandankumar4) [reply internal/util/kubernetes/unstructured_util.go:285]: Right

**Pull request merged**